addons/point_of_sale/static/src/app/generic_components/inputs/t_model_input.js

```diff
@@ -18,6 +18,6 @@ export class TModelInput extends Component {
     }
     setValue(newValue, tModel = this.props.tModel) {
         const [obj, key] = tModel;
-        obj[key] = newValue;
+        obj[key] = isNaN(newValue) ? 0 : newValue;
```

Comment:
the fix should be in the money_details_popup component: https://github.com/odoo/odoo/blob/b9f0f3676a9409a991ff8bd3d06ba4ace8d567e8/addons/point_of_sale/static/src/app/utils/money_details_popup/money_details_popup.js

Make sure that the registered count when it's empty is converted to zero.